Source/Core/Core/HW/EXI/BBA/BuiltIn.cpp

```diff
@@ -0,0 +1,686 @@
+  ref->my_mac = m_fake_mac;
+  ref->from.sin_addr.s_addr = 0;
+  ref->from.sin_port = htons(port);
+  ref->to.sin_addr.s_addr = m_current_ip;
+  ref->to.sin_port = htons(port);
+  ref->udp_socket.setBlocking(false);
+  if (ref->udp_socket.bind(port) != sf::Socket::Done)
+  {
+    ERROR_LOG_FMT(SP1, "Couldn't open UDP socket");
+    PanicAlertFmt("Could't open port {:x}, this game might not work proprely in LAN mode.", port);
+    return;
+  }
+}
+
+void CEXIETHERNET::BuiltInBBAInterface::HandleUDPFrame(const Common::UDPPacket& packet)
+{
+  const auto& [hwdata, ipdata, udpdata, ipoptions, data] = packet;
+  sf::IpAddress target;
+  const u32 destination_addr = ipdata.destination_addr == Common::IP_ADDR_ANY ?
+                                   m_router_ip :  // dns request
+                                   Common::BitCast<u32>(ipdata.destination_addr);
+
+  StackRef* ref = GetAvailableSlot(udpdata.source_port);
+  if (ref->ip == 0)
+  {
+    ref->ip = destination_addr;  // change for ip
+    ref->local = udpdata.source_port;
+    ref->remote = udpdata.destination_port;
+    ref->type = IPPROTO_UDP;
+    ref->bba_mac = Common::BitCastPtr<Common::MACAddress>(&m_eth_ref->mBbaMem[BBA_NAFR_PAR0]);
+    ref->my_mac = m_fake_mac;
+    ref->from.sin_addr.s_addr = destination_addr;
+    ref->from.sin_port = udpdata.destination_port;
+    ref->to.sin_addr.s_addr = Common::BitCast<u32>(ipdata.source_addr);
+    ref->to.sin_port = udpdata.source_port;
+    ref->udp_socket.setBlocking(false);
+    if (ref->udp_socket.bind(htons(udpdata.source_port)) != sf::Socket::Done)
+    {
+      PanicAlertFmt(
+          "Port {:x} is already in use, this game might not work as intented in LAN Mode.",
+          htons(udpdata.source_port));
+      if (ref->udp_socket.bind(sf::Socket::AnyPort) != sf::Socket::Done)
+      {
+        ERROR_LOG_FMT(SP1, "Couldn't open UDP socket");
+        return;
+      }
+      if (ntohs(udpdata.destination_port) == 1900)
+      {
+        InitUDPPort(26512);  // MK DD and 1080
+        InitUDPPort(26502);  // Air Ride
+        if (udpdata.length > 150)
+        {
+          // Quick hack to unlock the connection, throw it back at him
+          Common::UDPPacket reply = packet;
+          reply.eth_header.destination = hwdata.source;
+          reply.eth_header.source = hwdata.destination;
+          reply.ip_header.destination_addr = ipdata.source_addr;
+          if (ipdata.destination_addr == Common::IP_ADDR_SSDP)
+            reply.ip_header.source_addr = Common::IP_ADDR_BROADCAST;
+          else
+            reply.ip_header.source_addr = Common::BitCast<Common::IPAddress>(destination_addr);
+          WriteToQueue(reply.Build());
+        }
+      }
+    }
+  }
+  if (ntohs(udpdata.destination_port) == 53)
+    target = sf::IpAddress(m_dns_ip.c_str());  // dns server ip
+  else if (ntohl(Common::BitCast<u32>(ipdata.destination_addr)) ==
+           Common::BitCast<u32>(Common::IP_ADDR_SSDP))
```

Comment:
```C++
else if (ip.destination_addr == Common::IP_ADDR_SSDP)
```

Not to mention that `IP_ADDR_SSDP` is also using network-order _(since it's of type `IPAddress`/`std::vector<u8>`)_. So both operands should be converted to host-order or neither of them.